flixel/input/gamepad/id/XInputID.hx

```diff
@@ -4,52 +4,209 @@ import flixel.input.gamepad.FlxGamepad;
-	public static var RIGHT_ANALOG_STICK(default, null) = new FlxGamepadAnalogStick(2, 3);
-	
-	public static inline var LEFT_TRIGGER:Int = 4;
-	public static inline var RIGHT_TRIGGER:Int = 5;
-	
-	#if FLX_JOYSTICK_API
-	//the axis index values for this don't overlap with anything so we can just return the original values!
-	public static function axisIndexToRawID(index:Int):Int
-	{
-		return index;
-	}
+	#if flash
+		
+		// Button IDs
+		public static inline var A:Int = 4;
+		public static inline var B:Int = 5;
+		public static inline var X:Int = 6;
+		public static inline var Y:Int = 7;
+		public static inline var LB:Int = 8;
+		public static inline var RB:Int = 9;
+		
+		public static inline var BACK:Int = 12;
+		public static inline var GUIDE:Int = -1;
+		public static inline var START:Int = 13;
+		
+		public static inline var LEFT_STICK_CLICK:Int = 14;
+		public static inline var RIGHT_STICK_CLICK:Int = 15;
+		
+		public static inline var DPAD_UP:Int = 16;
+		public static inline var DPAD_DOWN:Int = 17;
+		public static inline var DPAD_LEFT:Int = 18;
+		public static inline var DPAD_RIGHT:Int = 19;
+		
+		// Axis indices
+		public static var LEFT_ANALOG_STICK(default, null) = new FlxGamepadAnalogStick(0, 1);
+		public static var RIGHT_ANALOG_STICK(default, null) = new FlxGamepadAnalogStick(2, 3);
+		
+		public static inline var LEFT_TRIGGER:Int = 10;
+		public static inline var RIGHT_TRIGGER:Int = 11;
+		
+		public static inline var LEFT_ANALOG_STICK_H = 0;
+		public static inline var LEFT_ANALOG_STICK_V = 1;
+		public static inline var RIGHT_ANALOG_STICK_H = 2;
+		public static inline var RIGHT_ANALOG_STICK_V = 3;
+		
+	#elseif FLX_GAMEINPUT_API
+		
+		// Button IDs
+		public static inline var A:Int = 6;
+		public static inline var B:Int = 7;
+		public static inline var X:Int = 8;
+		public static inline var Y:Int = 9;
+	
+		public static inline var BACK:Int = 10;
+		#if mac
+		public static inline var GUIDE:Int = 11;
+		#else
+		public static inline var GUIDE:Int = -1;
+		#end
+		public static inline var START:Int = 12;
+	
+		public static inline var LEFT_STICK_CLICK:Int = 13;
+		public static inline var RIGHT_STICK_CLICK:Int = 14;
+	
+		public static inline var LB:Int = 15;
+		public static inline var RB:Int = 16;
 	
-	//Just pass back LEFT/RIGHT trigger
-	public static inline var LEFT_TRIGGER_FAKE:Int = LEFT_TRIGGER;
-	public static inline var RIGHT_TRIGGER_FAKE:Int = RIGHT_TRIGGER;
+		public static inline var DPAD_UP:Int = 17;
+		public static inline var DPAD_DOWN:Int = 18;
+		public static inline var DPAD_LEFT:Int = 19;
+		public static inline var DPAD_RIGHT:Int = 20;
+	
+		// Axis indices
+		public static var LEFT_ANALOG_STICK(default, null) = new FlxGamepadAnalogStick(0, 1);
+		public static var RIGHT_ANALOG_STICK(default, null) = new FlxGamepadAnalogStick(2, 3);
+	
+		public static inline var LEFT_TRIGGER:Int = 4;
+		public static inline var RIGHT_TRIGGER:Int = 5;
+	
+		
+	#elseif FLX_JOYSTICK_API
+		
+		#if (windows || linux)
```

Comment:
Cool, I also think it's good to leave lists kind of "as-is" when large swaths are not obviously identical, because it makes it easier for a contributor to update an incorrect ID for a single platform without having to know what's going on with the other platforms.